build.gradle

```diff
@@ -1832,12 +1840,17 @@ project(':streams') {
 
     // testCompileOnly prevents streams from exporting a dependency on test-utils, which would cause a dependency cycle
     testCompileOnly project(':streams:test-utils')
+    // KAFKA-14109
+    // The below compileOnly dependency is needed for JUnit 4 tests.
+    // It can be safely removed once all of streams has moved to JUnit 5.
+    testCompileOnly libs.junit4
+
     testImplementation project(':clients').sourceSets.test.output
     testImplementation project(':core')
     testImplementation project(':core').sourceSets.test.output
     testImplementation libs.log4j
-    testImplementation libs.junitJupiterApi
-    testImplementation libs.junitVintageEngine
+    testImplementation libs.junitJupiter
+    testImplementation libs.junitJupiterParams // needed for parameterized tests
```

Comment:
I would prefer to add this when we need it during the migration of the parametrized tests. But I am also fine if it stays. 

The inline comment is not needed.

Reply:
I have removed this right now to keep this PR simple. Will add when needed.